shared/coordinator/src/coordinator.rs

```diff
@@ -1122,6 +1242,177 @@ impl<T: NodeIdentity> Coordinator<T> {
     pub fn is_training_just_starting(&self) -> bool {
         self.epoch_state.first_round.is_true() && self.run_state == RunState::RoundTrain
     }
+
+    fn get_consensus_on_reported_times(&self, witnesses: &[Witness]) -> Option<Vec<u16>> {
+        if witnesses.is_empty() {
+            return None;
+        }
+        if witnesses.len() == 1 {
+            return Some(witnesses[0].training_times.into());
+        }
+
+        let mut agreed_values = Vec::with_capacity(TRAINING_TIMES_SLICE_SIZE);
+
+        // For each time slot
+        for i in 0..TRAINING_TIMES_SLICE_SIZE {
+            let mut values: Vec<u16> = witnesses
+                .iter()
+                .map(|w| w.training_times.get(i).copied().unwrap_or(0))
+                .collect();
+
+            // If all values are 0, that's the consensus
+            if values.iter().all(|&v| v == 0) {
```

Comment:
Does this mean that the witnesses didn't receive any time for this client? Maybe we can do an enum like:

```rust
enum TrainingTimes {
     NotReceived,
     Received(u16)
}
```

And work with that, it will be better than checking 0 as the wrong value. I don't know if this will bring any problem to the witness message but we can think about it.